pkg/apis/pipeline/v1alpha1/type.go

```diff
@@ -0,0 +1,195 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced,categories=kurator-dev
+// +kubebuilder:subresource:status
+
+// Pipeline is the top-level type for Kurator CI Pipeline.
+type Pipeline struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   PipelineSpec   `json:"spec"`
+	Status PipelineStatus `json:"status,omitempty"`
+}
+
+// PipelineSpec defines the desired state of a Pipeline.
+type PipelineSpec struct {
+	// Description allows an administrator to provide a description of the pipeline.
+	// +optional
+	Description string `json:"description,omitempty"`
+
+	// Tasks is an ordered list of tasks in the pipeline, containing detailed information about each task.
+	// The tasks will be executed in the order they are listed.
+	Tasks []PipelineTask `json:"tasks"`
+
+	// SharedWorkspace is the name of the PVC. If not specified, a PVC with the Pipeline's name as prefix will be created by default.
+	// If not set, Kurator will create a PVC named Pipeline.name using default config
+	// +optional
+	SharedWorkspace *string `json:"sharedWorkspace,omitempty"`
+}
+
+type PipelineTask struct {
+	// Name is the name of the task.
+	Name string `json:"name"`
+
+	// TaskRef is a reference to a predefined task template.
+	// Users should provide a TaskRef name from a predefined library.
+	// +optional
+	TaskRef *TaskRef `json:"taskRef,omitempty"`
+
+	// CustomTask enables defining a task directly within the CRD if TaskRef is not used.
+	// This should only be used when TaskRef is not provided.
+	// +optional
+	CustomTask *CustomTask `json:"customTask,omitempty"`
+
+	// Retries represents how many times this task should be retried in case of task failure.
+	// +optional
+	Retries *int `json:"retries,omitempty"`
+}
+
+type PredefinedTask string
+
+const (
+	// GitClone is typically the first task in the entire pipeline.
+	// It clones the user's code repository into the workspace. This allows subsequent tasks to operate on this basis.
+	// Since the pipeline is linked to a specific repository's webhook, Kurator automatically retrieves the repository information when triggered by the webhook.
+	// Users don't need to configure additional repository information for this task, except for authentication details for private repositories.
+	// This Predefined Task is origin from https://github.com/tektoncd/catalog/tree/main/task/git-clone/0.9
+	// Here are the params that user can config:
+	// - git-secret-name: the secret name of git basic auth, Kurator use this git credential to clone private repo.
```

Comment:
how should user provide such secret and with what format

Reply:
we can add more format details, and tell user how to create it with following Pipeline doc